### PR TITLE
Added builder for list view

### PR DESCRIPTION
**Pull request by andr317c**

### CODE REVIEW

File: lib/builders/dataTypes/listViewDataTypeBuilder.ts [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    let values: any = [];
+
+    values.push({
+      alias: 'pageSize',
+      value: this.pageSize || 10
+    });
+
+    values.push({
+      alias: 'orderBy',
+      value: this.orderBy || "updateDate"
```
Comment (nhudinh0309) [inline lib/builders/dataTypes/listViewDataTypeBuilder.ts:82]:
```suggestion
      value: this.orderBy || "sortOrder"
```

File: lib/builders/dataTypes/listViewDataTypeBuilder.ts [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+      alias: 'bulkActionPermissions',
+      value: this.bulkActionPermissions.getValues()
+    });
+
+    values.push({
+      alias: 'icon',
+      value: this.icon || null
+    });
+
+    values.push({
```
Comment (nhudinh0309) [inline lib/builders/dataTypes/listViewDataTypeBuilder.ts]:
```suggestion
        if (this.tabName!== undefined) {
            values.push({
              alias: "tabName",
              value: this.tabName
            });
          }
```

File: lib/builders/dataTypes/listViewDataTypeBuilder.ts [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+      alias: 'tabName',
+      value: this.tabName || null
+    });
+
+    values.push({
```
Comment (nhudinh0309) [inline lib/builders/dataTypes/listViewDataTypeBuilder.ts]: Only push this variable when it is defined

File: lib/builders/dataTypes/listViewDataTypeBuilder.ts [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    this.bulkActionPermissions = builder;
+    return builder;
+  }
+
+  withIcon(icon: string) {
+    this.icon = icon;
+    return this;
+  }
+
+  withTabName(tabName: string) {
```
Comment (nhudinh0309) [inline lib/builders/dataTypes/listViewDataTypeBuilder.ts]: Would it be better to change the name to "withAppName" as in GUI, it is "Content app name"?

File: lib/builders/dataTypes/listViewBuilder/listViewPropertyBuilder.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
+﻿import {ListViewDataTypeBuilder} from "../listViewDataTypeBuilder";
+
+export class ListViewPropertyBuilder {
+  parentBuilder: ListViewDataTypeBuilder;
+  alias: string;
+  header: string;
+  nameTemplate: string;
```
Comment (nhudinh0309) [inline lib/builders/dataTypes/listViewBuilder/listViewPropertyBuilder.ts:7]: Missing the property named "isSystem"

Comment (nhudinh0309) [inline lib/builders/dataTypes/listViewBuilder/listViewPropertyBuilder.ts:7]: When the property "nameTemplate" is used? as I don't see it in json payload.

File: lib/builders/dataTypes/listViewDataTypeBuilder.ts [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+        return builder.getValues();
+      })
+    });
+
+    values.push({
+      alias: 'bulkActionPermissions',
+      value: this.bulkActionPermissions.getValues()
+    });
+
+    values.push({
```
Comment (nhudinh0309) [inline lib/builders/dataTypes/listViewDataTypeBuilder.ts:111]: Only push this variable when it is defined

**Review by nhudinh0309 — CHANGES_REQUESTED** [review]
I had some comments

File: lib/builders/dataTypes/listViewBuilder/listViewPropertyBuilder.ts (hunk 5 above)
Comment (andr317c) [reply lib/builders/dataTypes/listViewBuilder/listViewPropertyBuilder.ts:7]: It's in the columns displayed, if you add a value under "Enter a label template"

File: lib/builders/dataTypes/listViewDataTypeBuilder.ts (hunk 1 above)
Comment (andr317c) [reply lib/builders/dataTypes/listViewDataTypeBuilder.ts:82]: No reason to update this

**Comment by nhudinh0309** [comment]
LGTM

**Pull request merged**